hr_holidays_public/models/hr_leave.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2017-2018 Tecnativa - Pedro M. Baeza
+# Copyright 2018 Brainbean Apps
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class HrLeave(models.Model):
+    _inherit = 'hr.leave'
+
+    def _get_number_of_days(self, date_from, date_to, employee_id):
+        if (self.holiday_status_id.exclude_public_holidays or
+                not self.holiday_status_id):
+            instance = self.with_context(
+                employee_id=employee_id,
+                exclude_public_holidays=True,
+            )
+        else:
+            instance = self
+        return super(HrLeave, instance)._get_number_of_days(
+            date_from,
+            date_to,
+            employee_id,
+        )
+
+    def _onchange_employee_id(self):
+        super()._onchange_employee_id()
+        self._onchange_leave_dates()
+
+    @api.onchange('holiday_status_id')
+    def _onchange_holiday_status_id(self):
+        self._onchange_leave_dates()
```

Comment:
Why is this needed?

Reply:
And this is also needed or it is included upstream already?

Reply:
Indirectly, https://github.com/odoo/odoo/blob/bdfef60d60d78c5e7360fdabe80e70ea208e6078/addons/hr_holidays/models/hr_leave.py#L256
indirectly provokes `_onchange_request_parameters` via `_onchange_request_*`

What do yo think?